tests/e2e/run-e2e-tests.sh

```
@@ -0,0 +1,476 @@
+    else
+        test_failed "Concurrent requests failed"
+        return 1
+    fi
+}
+
+# Test: Agent cleanup on exit
+test_agent_cleanup() {
+    log_info "TEST: Agent cleanup on exit"
+
+    # Get initial session count from agent log
+    INITIAL_SESSIONS=$(grep -c "Creating session" "$TEST_RESULTS_DIR/agent.log" || echo 0)
```

Shellcheck (0.11.0)

[warning] INITIAL_SESSIONS appears unused. Verify use (or export if used externally).

(SC2034)